lake/index/match.go

```diff
@@ -0,0 +1,60 @@
+package index
+
+import (
+	"github.com/brimdata/zed"
+	"github.com/brimdata/zed/compiler/ast/dag"
+	"github.com/brimdata/zed/field"
+	"github.com/brimdata/zed/zson"
+)
+
+type match struct {
+	Rule       Rule
+	LookupKeys []*zed.Record
+}
+
+type predicate struct {
+	field field.Path
+	value zed.Value
+}
+
+type matcher func([]Rule) []match
```

Comment:
matcher wants to be close to its buddy, match.

```suggestion
type matcher func([]Rule) []match

type predicate struct {
	field field.Path
	value zed.Value
}
```